InnoViber.DAL/Data/ViberContext.cs

```diff
@@ -33,17 +34,20 @@ public override int SaveChanges()
 
         foreach (var entry in entries)
         {
-            var track = entry.Entity as BaseEntity;
+            var entity = entry.Entity as BaseEntity;
+
+            if (entity == null) { throw new FormatException("entity"); }
```

Comment:
1 Why have you chosen FormatException? (https://learn.microsoft.com/en-us/dotnet/api/system.formatexception?view=net-8.0)

2 You could just skip nullable entities